CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+
+1. Something is not working as described in the project.
+2. Something is not working as you expected it to work.
+3. Suggesting an idea that can improve the project.
+
+### Choosing an issue template
+
+Curve2Code provides two types of templates on the project's [issue page](https://github.com/svijaykoushik/curve2code/issues) to choose from
+
+1. Template for bug reporting and
+2. Template for suggesting enhancements and new ideas.
+
+You can choose the template which suits your issue(s) or choose to submit a regular issue template if your issue(s) don't fit in these two template.
+
+### Guidelines to open an issue
+
+Irrespective of the type of issue, all issues should follow these guidelines
+
+* Ensure that your issue is not already listed on the issue tracker page.
+* Provide a short and descriptive title for the issue.
+* Make a clear and concise description of the issue.
+* Make a clear and concise description of the impact of the issue on the project.
+
+## Submitting changes
+
+You can submit bug fixes and enhancements to the project by creating a "Pull request".
+
+### Before you start
+
+You must
+
+* Be familiar with cloning and branching git repositories.
+* Have knowledge in [GitHub flow](https://guides.github.com/introduction/flow/)
+* Posses knowledge in HTML, CSS & Javascript.
+* Be familiar with Javascript ES7 standards.
+* Have basic knowledge in Bezier curves.
+
+### Guidelines to submit changes
+
+* This project uses the [GitHub flow](https://guides.github.com/introduction/flow/) as the workflow.
+* Create a new branch from `master` to make your changes.
+* While providing branch names, use the following conventions
+  * If you are working on an issue that has already been logged then set the name as `issue-[provide issue number here]-[provide a short name for the issue here]`. eg: `issue-1-code-refactor`.
+  * If you are working on a bug or enhancement that has not yet been logged in the issue tracker then create a new issue and then follow the above mentioned guide line to set a name to the branch.
+
+* When working on an issue get the issue assigned to yourself from the project maintainer/owner and then start working on it.
+* Make commits of logical and atomic units.
+* Make sure your commit messages are in the proper format. If you're addressing an issue logged in the issue then provide the issue number on the front of the commit message with the parenthesis followed by the message indicating the changes made in a short and concise manner and in a commit description, providing a detailed description of the impact of the change(s) made has on the project.
+
+```
+    Commit message: (issue #1234) Adding contribution guidelines to the project`
+
+    Description: Without the contributing guidelines the contributors would confused on how to get started on contributing to this project. This patch would help contributors gain knowledge on the internal workflow and project guidelines to be followed.
```

Comment:
change to: Without the contributing guidelines the contributors would have no idea on how to get started to contribute to this project.